### PR TITLE
Feat(Forms/Select): Split Select into Select and MultiSelect

### PR DESCRIPTION
**Pull request by RostiMelk**
To avoid confusion in return values, the Select component has been split into two components; Select and MultiSelect.
The component has also undergone some cleanup and fixes to avoid unnecessary re-renders.

### CODE REVIEW

File: src/stories/Forms/MultiSelect/MultiSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+import { GlobalInputProps } from '../../../shared/types';
+import { OptionProps } from './Option';
+import { Search } from '@huddly/frokost/havre';
+import * as Styled from '../Select/styled';
+
+// Custom hooks
+import { useOnClickOutside } from '../../../shared/hooks/index';
+
+// Utility functions
+import {
```
Comment (amfestoy) [inline src/stories/Forms/MultiSelect/MultiSelect.tsx:12]: Liker at disse ble tatt ut i egen fil 😁

File: src/stories/Forms/MultiSelect/MultiSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+      };
+    }, [isOpen]);
+
+    const handleKeyboardEvent = (event: KeyboardEvent) => {
+      const selectList = selectListRef.current;
+      const selectListChildren = selectList?.children as HTMLCollectionOf<HTMLElement>;
+      const activeElement = document.activeElement as HTMLElement;
+      const activeOptionIndex = Array.from(selectListChildren).indexOf(activeElement);
+
+      const keyActions = {
```
Comment (amfestoy) [inline src/stories/Forms/MultiSelect/MultiSelect.tsx:94]: This is clean and nice af

File: src/stories/Forms/Select/Option.tsx [hunk 3]
```diff
@@ -11,6 +11,12 @@ const Wrapper = styled.li`
   outline: 0;
   column-gap: var(--spacing-8);
   cursor: pointer;
+  transition: background-color 0.2s ease-in-out;
+
+  /* &:hover,
```
Comment (amfestoy) [inline src/stories/Forms/Select/Option.tsx:16]: forgot to delete commented out code?

File: src/stories/Forms/Select/styled.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+  &,
+    & > * {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+`;
+
+export const RotatingChevron = styled(ChevronDown)<{ rotate: boolean }>`
```
Comment (amfestoy) [inline src/stories/Forms/Select/styled.ts:71]: Are you sure this is not a carrot? 🐇

**Review by amfestoy — APPROVED** [review]
This is well written and easy to read :D

File: src/stories/Forms/Select/styled.ts (hunk 4 above)
Comment (RostiMelk) [reply src/stories/Forms/Select/styled.ts:71]: [maybe](https://www.gingersoftware.com/english-online/spelling-book/confusing-words/carat-caret-karat-carrot)

**Pull request merged**